pkg/api/v1/types_kubelet.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api/resource"
+)
+
+type NodeStats struct {
+	TypeMeta
+	ObjectMeta // should have node name
+	Containers []struct {
+		Name  string            `json:"name"`
+		Stats []*ContainerStats `json:"stats"`
+	}
+}
+
+type ContainerStats struct {
+	Time time.Time `json:"time"`
+
+	Cpu    CpuStats    `json:"cpu"`
+	Memory MemoryStats `json:"memory"`
```

Comment:
Do expose all the information that is currently available via the '/stats' endpoint.